catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +170,54 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService("dog_legs.sch", "dog_paws.sch");
+        // test warnings
+        try {
+            service.validate(getMetacard("dog_4leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String warning : ex.getWarnings()) {
+                assertThat(warning.equals(SchematronValidationService.sanitize(warning)), is(true));
+            }
+        }
+        // test errors
+        try {
+            service.validate(getMetacard("dog_3leg_3paw.xml"));
+        } catch (SchematronValidationException ex) {
+            assertThat(service.getSchematronReport(), is(notNullValue()));
+            for (String error : ex.getErrors()) {
+                assertThat(error.equals(SchematronValidationService.sanitize(error)), is(true));
+            }
+        }
+        // sanitize is called twice for each validate call
+        verify(service, times(4)).sanitize(anyString());
+    }
+
+    @Test
+    public void testSanitizationChangesNothing() {
+        // verify that nothing changes
+        assertThat("ontattoinewerunfromsandpeople",
+                is(SchematronValidationService.sanitize("ontattoinewerunfromsandpeople")));
+        // verify that nothing changes again
+        assertThat("princess leia", is(SchematronValidationService.sanitize("princess leia")));
+    }
+
+    @Test
+    public void testSanitizationChangesInput() throws SchematronInitializationException {
```

Comment:
I don't think this test should throw a SchematronInitializationException if the other one doesn't either, right?